package.json

```diff
@@ -124,6 +124,10 @@
     "run-script-os": "^1.1.6",
     "typescript": "~4.5.5"
   },
+  "engines": {
+    "node": ">=18.17.1",
+    "npm": "^6.14.18"
+  },
```

Comment:
Still can't get this to ^9?

Reply:
And it we can't you'll need to add notes to readme because node 18 w/o npm 9 is an odd ball that needs mentioned.

Reply:
The problem is the in CI there are policy checks for the lock files to be generated with npm6

Reply:
We can change the policy check - we are changing node to get to stable tools.

Reply:
I think Craig is right that you should revert the package-lock.json changes if you are keeping npm 6.
I don't know that it is okay to stay on npm 6. Is it not going out of LTS like old node? But that doesn't mean it has to be this PR.
Thanks for taking up the mantle. It can feel unhelpful and very much a tax. It will be good to get to 2.0 and hopefully we'll have better process to keep things healthier.

@tylerbutler, shouldn't we be in control of all of the policy checks?

Reply:
> @tylerbutler, shouldn't we be in control of all of the policy checks?

Alas, policy-check was not configurable in the past. All the handlers are always run. I added [configurability a while back](https://github.com/microsoft/FluidFramework/pull/12648) but that change wasn't made until build-tools 0.13.0. The version of build-tools used in lts (0.6.0) can be patched to remove the policy-handler. The change should be made in the release/build-tools/0.6 branch.

That said, I think that change can and should be done separately, especially since this PR seems very close to being mergeable.

Reply:
Any reason not to set >=6 instead of restricting to 6? (I am perfectly cool with ^6 for now if that accelerates merge.)

Reply:
> Any reason not to set >=6 instead of restricting to 6?

The reason the policy existed before was because we used node 12 in the repo, which came with npm 6, but node 14 came with npm 7 by default AND it used a new lockfile format. So people that were using 14+ would mistakenly update the lockfile format in their PRs, causing flip-flops in the format and a bunch of unneeded lockfile changes. Now that we support node 18, and the lockfile format has been stable (AFAIK) since npm 7, there's no reason to keep the branch on npm 6. It's fine to update in a follow-up PR with the build-tools patch I described.

Note that we will likely need to use some settings like legacy-peer-deps because lts is NOT going to be easy to make behave well with modern behaviors. Example: https://github.com/microsoft/FluidFramework/issues/8117

Reply:
The lock files were generated using npm6 when I ran `npx npm@6.14.18 i`, so I think they should be included, shouldn't they?

Reply:
Well, that is a command that asks for dependencies to be changed. We shouldn't change things that we aren't intending to change (general principle).
If you revert `package-lock.json`, run `npm ci`, and all is fine, then that is what is preferred.
npm i or npm install is free to update dependencies and rewrite the lock file. npm ci keeps package-lock.json intact.
I only gave the dependency changes a quick glance because I wasn't going to scrutinize if dep updates are not intended or not using the version on npm preferred. I do see that some dependencies have gone backwards for better or worse. I prefer not to think about them (revert), but I can ask about why there are certain changes if you want to refresh dependencies.